test/Test/GameDAO.hs

```diff
@@ -0,0 +1,136 @@
+-- SPDX-FileCopyrightText: 2020 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+module Test.GameDAO
+  ( test_GameDAO
+  ) where
+
+import Universum hiding (compare, drop, (>>))
+
+import Lorentz
+import Lorentz.Test (contractConsumer)
+import Morley.Nettest
+import Morley.Nettest.Tasty
+import Test.Tasty (TestTree, testGroup)
+import Time (sec)
+
+import qualified Lorentz.Contracts.BaseDAO.Types as DAO
+import Lorentz.Contracts.GameDAO
+import Test.Common
+
+{-# ANN module ("HLint: ignore Reduce duplication" :: Text) #-}
+
+test_GameDAO :: TestTree
+test_GameDAO = testGroup "GameDAO Tests"
+  [ testGroup "Proposal creator:"
+      [ nettestScenario "can propose a valid proposal" validProposal
+      ]
+  , testGroup "Admin"
+      [ nettestScenario "can flush proposals that got accepted" flushAcceptedProposals
+      ]
+  ]
+
+validProposal :: (Monad m) => NettestImpl m -> m ()
+validProposal = uncapsNettest $ do
+  (consumer :: TAddress MText) <- originateSimple "consumer" [] contractConsumer
+  ((owner1, _), _, dao, _) <- originateBaseDaoWithConfig config
+
+  -- Fail due to proposing new content require 50 token.
+  callFrom (AddressResolved owner1) dao (Call @"Propose") (DAO.ProposeParams
+    { ppFrozenToken = 20
+    , ppProposalMetadata = sampleMetadataContent $ toAddress consumer
+    }) & expectCustomError_ #fAIL_PROPOSAL_CHECK
+
+  callFrom (AddressResolved owner1) dao (Call @"Propose") (DAO.ProposeParams
+    { ppFrozenToken = 15
+    , ppProposalMetadata = sampleMetadataBalance $ toAddress consumer
+    })
+
+  checkTokenBalance (DAO.frozenTokenId) dao owner1 15
+  checkTokenBalance (DAO.unfrozenTokenId) dao owner1 85
+
+flushAcceptedProposals :: (Monad m) => NettestImpl m -> m ()
+flushAcceptedProposals = uncapsNettest $ do
+  (consumer :: TAddress MText) <- originateSimple "consumer" [] contractConsumer
+  ((owner1, _), (owner2, _), dao, admin)
+    <- originateBaseDaoWithConfig config
+
+  callFrom (AddressResolved admin) dao (Call @"Set_voting_period") 60
+  callFrom (AddressResolved admin) dao (Call @"Set_quorum_threshold") 1
+
+  -- | Accepted Proposals
+  key1 <- createSampleProposal (sampleMetadataBalance $ toAddress consumer) owner1 dao
+  checkTokenBalance (DAO.frozenTokenId) dao owner1 10
+
+  let upvote = DAO.VoteParam
+        { vVoteType = True
+        , vVoteAmount = 2
+        , vProposalKey = key1
+        }
+      downvote = DAO.VoteParam
+        { vVoteType = False
+        , vVoteAmount = 1
+        , vProposalKey = key1
+        }
+  callFrom (AddressResolved owner2) dao (Call @"Vote") [upvote, downvote]
+  checkTokenBalance (DAO.frozenTokenId) dao owner2 3
+  checkTokenBalance (DAO.unfrozenTokenId) dao owner2 97
+
+  advanceTime (sec 61)
```

Comment:
If I understand correctly, our CI will run this with tezos-client, and `61` will mean that this test takes a minute. Maybe reduce the voting period a bit to make CI faster?

Reply:
It was made this way due to in real dephi network tests, there can be a lot of delay. However, we found out that it won't be helpful anyway and only enable all time-sensitive tests (flush) in emulator. 

So I think I'll reduce it to smaller number. When we are able to run operations in batch, we should be able enable these tests in real network.

Reply:
Oh, btw, it seems that currently this test runs not only in emulator. I think we should change it here as we did in #44.

Reply:
I forgot about this. I've made so that it will run only in the emulator instead now.